app/controllers/api/v1/orders_controller.rb

```diff
@@ -51,31 +51,37 @@ def pay
       )
 
       if charge.status == "succeeded" && charge.paid
-        @order.update(stripe_charge_id: charge.id)
+        @order.update(stripe_charge_id: charge.id, status: "processed")
       else
         unsuccesful_payment
       end
+
     rescue Stripe::CardError => e
-      unsuccesful_payment e.json_body[:error]
+      error = e.json_body[:error] if e.json_body
+      unsuccesful_payment error
     rescue Stripe::InvalidRequestError => e
-      unsuccesful_payment e.json_body[:error]
+      error = e.json_body[:error] if e.json_body
+      unsuccesful_payment error
     rescue Stripe::AuthenticationError => e
-      Rails.logger.error e.json_body[:error]
+      error = e.json_body[:error] if e.json_body
+      unsuccesful_payment error
       raise
     rescue Stripe::StripeError => e
-      unsuccesful_payment
+      error = e.json_body[:error] if e.json_body
```

Comment:
well this is related to the error message from stripe, I think it should stay here.

I tried to use only one `rescue` but it didn't work.
like `rescue Stripe::StripeError => e, Stripe::CardError => e ...`